### PR TITLE
feat: sidebar added with routing implementation

### PR DESCRIPTION
**Pull request by pankaj-prog**
Create sidebar and implemented routing
BlitheBrains - sidebar

Files to check: 
1. App.js
2. Sidebar.js
3. Index.js
4. Sidebar components

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,7 +1,47 @@
...
+          <Route path="/home" element={<Home />} />
+          <Route path="/explore" element={<Explore />} />
+          <Route path="/history" element={<History />} />
+          <Route path="/likedvideos" element={<LikedVideos />} />
+          <Route path="/playlist" element={<Playlist />} />
+          <Route path="/watchlater" element={<WatchLater />} />
+          <Route path="/login" element={<Login />} />
+          <Route path="/signup" element={<Signup />} />
+          <Route path="*" element={<Page404 />} />
+        </Routes>
```
Comment (PuneetP16) [inline src/App.js]: Suggestion: you could wrap this up in a  Routes component.

File: src/components/Sidebar/components/NavItemExplore.css [hunk 2]
```diff
@@ -0,0 +1,11 @@
+.sidebar .categories-section {
+  display: none;
+  list-style-type: none;
+  padding: 0.5rem 0 0 3rem;
+}
+
```
Comment (PuneetP16) [inline src/components/Sidebar/components/NavItemExplore.css:6]: Suggestion: you could change this to px as this values are too small to factor in responsiveness using rems

File: src/components/Sidebar/components/NavItemExplore.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import "./NavItemExplore.css";
+import { NavLink } from "react-router-dom";
+import { MdOutlineExplore } from "react-icons/md";
+
+const NavItemExplore = () => {
+  return (
+    <NavLink className="link " to="explore">
+      {({ isActive }) => (
+        <>
```
Comment (PuneetP16) [inline src/components/Sidebar/components/NavItemExplore.jsx:11]: Appreciation: liked that you are using Navlink and its active styling facility.

File: src/components/Sidebar/Sidebar.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        </h2>
+        <button
+          onClick={() => setShowUserWrapper(!showUserWrapper)}
+          class="avatar avatar-sm avatar-round"
+        >
+          <img
+            class="responsive-img"
+            src={images.defaultAvatar}
+            alt="default avatar"
+          />
```
Comment (PuneetP16) [inline src/components/Sidebar/Sidebar.jsx:25]: UI Suggestion: You could increase the size of the avatar a little bit.

image

File: src/components/Sidebar/components/NavigationMenu.jsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+const navItems = [
+  { name: "Home", path: "/home", icon: <AiOutlineHome /> },
+  { name: "Explore" },
+  { name: "Liked Videos", path: "/likedvideos", icon: <AiOutlineLike /> },
+  { name: "Playlist", path: "/playlist", icon: <MdPlaylistPlay /> },
+  { name: "Watch Later", path: "/watchlater", icon: <MdOutlineWatchLater /> },
+  { name: "History", path: "/history", icon: <AiOutlineHistory /> },
+];
+
```
Comment (PuneetP16) [inline src/components/Sidebar/components/NavigationMenu.jsx:18]: Appreciation: Good work on creating navItems list and rendering the navbar through it.

File: src/App.js [hunk 6]
```diff
@@ -1,7 +1,47 @@
...
 
 function App() {
-  return <div className="App">APP</div>;
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    location.pathname == "/" ? navigate("/home") : undefined;
+  }, [location]);
+
```
Comment (manojsarna) [inline src/App.js]: Suggestion use && if you have only one condition to handle

**Review by PuneetP16 — APPROVED** [review]
check the comments mentioned. good to merge.

File: src/components/Sidebar/components/NavigationMenu.jsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import { AiOutlineHome, AiOutlineLike, AiOutlineHistory } from "react-icons/ai";
+import { MdOutlineWatchLater, MdPlaylistPlay } from "react-icons/md";
+
+import { NavLink } from "react-router-dom";
+import NavItemExplore from "./NavItemExplore";
+
+const navItems = [
+  { name: "Home", path: "/home", icon: <AiOutlineHome /> },
+  { name: "Explore" },
```
Comment (manojsarna) [inline src/components/Sidebar/components/NavigationMenu.jsx:12]: Is this alright to include icons components in object, need to check

File: src/components/Sidebar/components/NavItemExplore.jsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+          <ul
+            className={`categories-section  ${
+              isActive ? "categories-section-active" : undefined
+            }`}
+          >
+            <li>Podcast</li>
+            <li>Book Summary</li>
+            <li>Case study</li>
+          </ul>
+        </>
```
Comment (manojsarna) [inline src/components/Sidebar/components/NavItemExplore.jsx:30]: Suggestion : Get all the categories from backend APIs

File: src/components/Sidebar/components/NavItemExplore.jsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+import "./NavItemExplore.css";
+import { NavLink } from "react-router-dom";
+import { MdOutlineExplore } from "react-icons/md";
+
+const NavItemExplore = () => {
+  return (
+    <NavLink className="link " to="explore">
+      {({ isActive }) => (
+        <>
+          <span
```
Comment (manojsarna) [inline src/components/Sidebar/components/NavItemExplore.jsx:12]: Appreciation : good that you are using active links

**Review by manojsarna — APPROVED** [review]
overall it looks good!!

Do check the comments

**Review by m26unkwn — APPROVED** [review]
EVERYTHING IS GOOD  AND FINE.
Wonderful UI superb.

1. using array objects for navitems

File: src/App.js [hunk 10]
```diff
@@ -1,7 +1,47 @@
...
+  WatchLater,
+} from "./pages";
 
 function App() {
-  return <div className="App">APP</div>;
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    location.pathname == "/" ? navigate("/home") : undefined;
```
Comment (bhtibrewal) [inline src/App.js]: Why have you given undefined here, you could have done like:
```
(location.pathname == "/")  && navigate("/home")

**Review by bhtibrewal — APPROVED** [review]
Couldn't find any mistake, could code, appreciate the way u thought about making everything modular just, it would be better if you could also write comment where the code goes a little complex, good to merge

File: src/components/Sidebar/components/NavigationMenu.jsx (hunk 7 above)
Comment (pankaj-prog) [reply src/components/Sidebar/components/NavigationMenu.jsx:12]: yeah this is working fine

File: src/components/Sidebar/components/NavigationMenu.jsx (hunk 5 above)
Comment (pankaj-prog) [reply src/components/Sidebar/components/NavigationMenu.jsx:18]: Thankyou

File: src/components/Sidebar/Sidebar.jsx (hunk 4 above)
Comment (pankaj-prog) [reply src/components/Sidebar/Sidebar.jsx:25]: I will try it out

File: src/App.js (hunk 10 above)
Comment (pankaj-prog) [reply src/App.js]: will fix this

**Pull request merged**